src/main/java/com/immunopass/WebApp.java

```diff
@@ -0,0 +1,31 @@
+package com.immunopass;
+
+import com.immunopass.interceptor.AuthInterceptor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@EnableWebMvc
+@Configuration
+public class WebApp implements WebMvcConfigurer {
+
+    public static List<String> AUTH_EXCLUDE_PATTERNS = new ArrayList();
+
+    @Bean
+    public AuthInterceptor getAuthInterceptor() {
+        return new AuthInterceptor();
+    }
+
+    @Override
+    public void addInterceptors(InterceptorRegistry registry) {
+        registry.addInterceptor(getAuthInterceptor()).
+                addPathPatterns("/**").
+                excludePathPatterns(AUTH_EXCLUDE_PATTERNS);
```

Comment:
We can exclude OTP generation part, otherwise we won't be able to trigger OTP.

Reply:
Yeah, I didn't add the end point that we have right now.. because the otp end point that we have right now.. is doing multiple operations.. wanted to be sure if we wanted to move forward like that before adding the endpoint here..

Ideally, those API end point should be different specifically.. verify and create otp.. as their response will be completely different.

Reply:
Sure. Let's discuss on the OTP endpoints.